warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java

```diff
@@ -169,9 +171,15 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
     BigInteger rinv = spec.getCurve().fromBigInteger(r.modInverse(spec.getN())).toBigInteger();
 
+    int minH = 0;
     int maxH = H.intValue();
 
-    if (maxH > MAX_COFACTOR) {
+    if (params.get(KEY_I) instanceof Long) {
+      minH = ((Long) params.get(KEY_I)).intValue();
+      maxH = minH;
```

Comment:
There must be a difference of 1 between those two for the `for` loop to execute. I'd do `minH = ((Long) params.get(KEY_I)).intValue() - 1;`.

Reply:
It's better if maxH is set to minH + 1, because the loop starts at minH.